flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/service/StandaloneFlinkServiceTest.java

```diff
@@ -177,11 +177,123 @@ public void testReactiveScale() throws Exception {
                 .getSpec()
                 .getFlinkConfiguration()
                 .remove(JobManagerOptions.SCHEDULER_MODE.key());
-        assertFalse(
+        assertTrue(
+                flinkStandaloneService.scale(
+                        flinkDeployment.getMetadata(),
+                        flinkDeployment.getSpec().getJob(),
+                        buildConfig(flinkDeployment, configuration)));
+    }
+
+    @Test
+    public void testTMReplicaScale() throws Exception {
+        var flinkDeployment = TestUtils.buildApplicationCluster();
```

Comment:
would be nice to test this with a session cluster resource where `job = null`.
You could use `TestUtils.buildSessionCluster()` instead